lib/fake_ifdh.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python3 -I
+
+# fake_ifdh -- get rid of ifdhc dependency by providing a few
+#              bits of ifdh behavior
+#
+# COPYRIGHT 2021 FERMI NATIONAL ACCELERATOR LABORATORY
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+import os
+import time
+import argparse
+
+VAULT_HOST = "fermicloud543.fnal.gov"
+DEFAULT_ROLE = "Analysis"
+
+
+def getTmp():
+    return os.environ.get("TMPDIR", "/tmp")
+
+
+def getExp():
+    for ev in ["GROUP", "EXPERIMENT", "SAM_EXPERIMENT"]:
+        if os.environ.get(ev, None):
+            return os.environ.get(ev)
+    # otherwise guess primary group...
+    f = os.popen("id -gn", "r")
+    exp = f.read()
+    f.close()
+    return exp
+
+
+def getRole(role_override=None):
+    if role_override:
+        return role_override
+    elif os.environ["USER"][-3:] == "pro":
+        return "Production"
+    else:
+        return DEFAULT_ROLE 
+
+
+def checkToken(tokenfile):
+    f = os.popen("decode_token.sh -e exp %s 2>/dev/null" % tokenfile, "r")
+    exp_time = f.read()
+    f.close()
+    return  exp_time and ((int(exp_time) - time.time()) > 60)
+
+def getToken(role=DEFAULT_ROLE):
+    pid = os.getuid()
+    tmp = getTmp()
+    exp = getExp()
+    if exp == "samdev":
+        issuer = "fermilab"
+    else:
+        issuer = exp
+    tokenfile = "%s/bt_token_%s_%s_%s" % (tmp, issuer, role, pid)
+    os.environ["BEARER_TOKEN_FILE"] = tokenfile
+    if not checkToken(tokenfile):
```

Comment:
So I've always preferred the style of:

```
if what_I_want:
    do_something()
else:
    do_something_else()
```

What would you think about changing these lines to:

```
if checkToken(tokenfile):
    return tokenfile
else:
    all_the_other_stuff
```
? 

Not something I'm adamant on changing - I'm just curious if you agree.

Reply:
It's all equivalent; that's just what I was thinking when I wrote it -- I do the work of getting a token if the checkToken fails.